tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -384,22 +384,36 @@ struct ParentInfo {
   StringRef USR;
 };
 
+struct CursorSymbolInfo {
+  UIdent Kind;
+  UIdent DeclarationLang;
+  llvm::SmallString<32> Name;
```

Comment:
All used to be StringRef, but that gets quite messy when we have multiple results - we'd have to keep begin/end for all these fields and then assign it all once the buffer is stable.

There's not *that* many string here so that didn't really seem worth it to me.

CursorInfoData still has a bunch of StringRefs that do this still, happy to either do them in this PR, do them in another PR, or just leave until later. Alternatively, let me know if you think I should do something else here. It *could* use the buffer if it has to.

Reply:
> but that gets quite messy when we have multiple results

What's the issue here? We're sticking them all into a string stream that lives only for the callback so I would have expected it to be equivalent whether there's one of many.

Reply:
The issue is that we can't add a StringRef for anything in the stream until the buffer is stable, ie. until everything has been added. That's fine-is when there's only a single result, it current just keeps the start/end for each field and then sets all the StringRefs at the end.

Keeping track of it for multiple results is possible, just wasn't sure it was worth it. But you're right, the mix of StringRef/SmallString is a bit confusing. I'll try a few other options and see how it feels.